src/renderer/components/Login.vue

```diff
@@ -1,19 +1,110 @@
 <template>
-  <el-card>
-    <img id="logo" src="~@/assets/logo.svg" alt="Iroha">
-    <span>Login page, Welcome!</span>
+  <el-card class="login-form-container">
+    <img id="logo" class="logo" src="~@/assets/logo.svg" alt="Iroha">
+
+    <el-form ref="form" :model="form" :rules="rules" label-position="top">
+      <el-form-item label="username:" prop="username">
+        <el-input name="username" v-model="form.username"></el-input>
+      </el-form-item>
+
+      <el-form-item label="private key:" prop="privateKey">
+        <el-row type="flex" justify="space-between">
+          <el-col :span="17">
+            <el-input name="privateKey" v-model="form.privateKey"></el-input>
+          </el-col>
+
+          <el-col :span="6">
+            <el-upload
+              action=""
+              :auto-upload="false"
+              :show-file-list="false"
+              :on-change="onFileChosen"
+            >
+              <el-button type="primary">choose</el-button>
+            </el-upload>
+          </el-col>
+        </el-row>
+      </el-form-item>
+
+      <el-form-item label="node ip:" prop="nodeIp">
+        <el-input v-model="form.nodeIp"></el-input>
+      </el-form-item>
+
+      <el-form-item>
+        <el-button type="primary" @click="onSubmit">Login</el-button>
+      </el-form-item>
+    </el-form>
   </el-card>
 </template>
 
 <script>
   export default {
-    name: 'login'
+    name: 'login',
+
+    data () {
+      return {
+        form: {
+          username: '',
+          privateKey: '',
+          nodeIp: ''
+        },
+        rules: {
+          username: [
+            { required: true, message: 'Please input username', trigger: 'change' },
+            { pattern: /^[a-z_0-9]{1,32}@[a-z_0-9]{1,9}$/, message: 'Username should match [a-Z_0-9]{1,32}@[a-Z_0-9]{1,9}', trigger: 'change' }
+          ],
+          privateKey: [
+            { required: true, message: 'Please input private key', trigger: 'change' },
+            { pattern: /^[A-Za-z0-9]{64}$/, message: 'Private key should match [A-Za-z0-9]{64}', trigger: 'change' }
+          ],
+          nodeIp: [
+            { required: true, message: 'Please input node ip', trigger: 'change' }
```

Comment:
Why no IP validation? :( However I guess it is debatable since there could be virtually any valid domain name, so maybe there should be no validation.

Reply:
As I mentioned in the description, the regex pattern to validate both IP and a domain name was too complicated, so I didn't put it here on purpose. I added IP validation.